drivers/builtin/src/md.c

```diff
@@ -112,25 +112,25 @@ static const mbedtls_md_info_t mbedtls_sha512_info = {
 };
 #endif
 
-#if defined(PSA_WANT_ALG_SHA3_224)
+#if defined(PSA_WANT_ALG_SHA3_224) || defined(MBEDTLS_PSA_ACCEL_ALG_SHA3_224)
```

Comment:
I don't understand the motivation for this change. I was under the impression that it's not possible to have `MBEDTLS_PSA_ACCEL_ALG_xxx` defined without `PSA_WANT_ALG_xxx` (at least, we don't care about working in such configs, and shouldn't generate such configs in tests).

Cc @gilles-peskine-arm

Reply:
> shouldn't generate such configs in tests

Even with libtestdriver1? It does funky undocumented things with the configuration, especially wrt hashes (accelerating an algorithm that uses hashes without accelerating the hashes themselves, or vice versa), and I don't remember offhand how everything works.

Reply:
Ah, perhaps when we build the libtestdriver1 part indeed, where we might do things no normal user should ever do.

Reply:
This test made me to add the change:
https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/7743195de4744b9cbe870ac22aad671a9188ff81/tests/suites/test_suite_md.psa.data#L93-L95

Reply:
Ok, I'm glad I asked because these test dependencies are incorrect and I failed to notice earlier. Here we don't want `!PSA_WANT_ALG_SHA3_224` but instead `!MBEDTLS_PSA_BUILTIN_ALG_SHA3_224`. The test is about having this alg only through a driver, so we want to express that we don't have the built-in implementation, not that we don't have the alg.

Reply:
I'm still unconvinced this change was needed, but it's not harmful, and all of this should disappear at some point when we made MD a thin wrapper around `psa_hash`, so let's not waste time fine-tuning code that's going to be removed. I'm marking this conversation as resolved.